### ISSUE TITLE
Add built-in struct types for indirect calls

### ISSUE THREAD

**Issue by teoxoy** [gpuweb/gpuweb#4158]
It would be useful to add the following built-in struct types since users would have to define those themselves anyway if they want to write the parameters to an indirect buffer.

```rust
struct DrawIndirectParameters {
    vertexCount: u32,
    instanceCount: u32,
    firstVertex: u32,
    firstInstance: u32,
}
```

```rust
struct DrawIndexedIndirectParameters {
    indexCount: u32,
    instanceCount: u32,
    firstIndex: u32,
    baseVertex: u32,
    firstInstance: u32,
}
```

```rust
struct DispatchWorkgroupsIndirectParameters {
    workgroupCountX: u32,
    workgroupCountY: u32,
    workgroupCountZ: u32,
}
```

_originally requested by @JMS55 in https://github.com/gfx-rs/naga/issues/2360_

**Comment by teoxoy** [comment]
Want to point out that the struct names and their respective member names came from what we currently call them in the spec (struct names have an additional "Parameters" suffix). Ex:

![image](https://github.com/gpuweb/gpuweb/assets/28601907/cb402d0b-74c4-4385-8e93-c33a8f2eb8d8)

**Comment by dneto0** [comment]
@toji mentioned that often we want the count fields to be atomic.

**Comment by JMS55** [comment]
Yeah I can't remember my original use case, but anything I've used indirect draws/dispatches for recently, I've wanted one or more fields to be atomic.
